### PR TITLE
Added XSOAR-linter to pre-commit

### PR DESCRIPTION
**Pull request by dansterenson**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-7062

Content related PR: https://github.com/demisto/content/pull/32302

Reduced the lint process to 50 minutes (instead of ~3 hours) since now it uses multiprocessing:
Here is an example where all errors are collected at the end:

image

https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/jobs/4500546

### CODE REVIEW

File: demisto_sdk/commands/pre_commit/hooks/xsoar_linter.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        Prepares the Xosar-linter hook for each Python version.
+        Changes the hook's name, files and the "--target-version" argument according to the Python version.
+        Args:
+        """
+        for integration_script_obj, files in self.context.object_to_files.items():
+            xsoar_linter_env = {}
+            if isinstance(integration_script_obj, Integration) and integration_script_obj.long_running:
+                xsoar_linter_env["LONGRUNNING"] = "True"
+
+            if (py_ver := integration_script_obj.python_version) and Version(py_ver).major < 3:
```
Comment (dorschw) [inline demisto_sdk/commands/pre_commit/hooks/xsoar_linter.py:32]: If one content item is py2, we'll have this env var set for all items?

File: demisto_sdk/commands/pre_commit/hooks/xsoar_linter.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            hook.update({
+                "name": f"xsoar-linter-{integration_script_obj.object_id}",
+                "args": args,
+                "entry": f'env {xsoar_linter_env_str} {Path(sys.executable).parent}/{self.base_hook["entry"]}'
+            })
+            hook["files"] = join_files(
+                {
+                    file
+                    for file in files
+                    if file.suffix == ".py" and '_test' not in file.name
```
Comment (dorschw) [inline demisto_sdk/commands/pre_commit/hooks/xsoar_linter.py:56]:
```suggestion
                    if file.suffix == ".py" and not file.name.endswith('_test')
```

File: demisto_sdk/commands/xsoar_linter/xsoar_linter.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+from demisto_sdk.commands.lint.resources.pylint_plugins.base_checker import base_msg
+from demisto_sdk.commands.lint.resources.pylint_plugins.certified_partner_level_checker import cert_partner_msg
+from demisto_sdk.commands.lint.resources.pylint_plugins.community_level_checker import community_msg
+from demisto_sdk.commands.lint.resources.pylint_plugins.partner_level_checker import partner_msg
+from demisto_sdk.commands.lint.resources.pylint_plugins.xsoar_level_checker import xsoar_msg
+from demisto_sdk.commands.content_graph.objects import Integration, Script
+from demisto_sdk.commands.content_graph.objects.base_content import BaseContent
+from demisto_sdk.commands.content_graph.objects.integration_script import IntegrationScript
+
+xsoar_linter_app = typer.Typer(name="Pre-Commit")
```
Comment (ilaner) [inline demisto_sdk/commands/xsoar_linter/xsoar_linter.py:24]: please remove

File: demisto_sdk/commands/xsoar_linter/xsoar_linter.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    # Enable only Demisto Plugins errors.
+    command.append(f"--enable={message_enable}")
+    # Load plugins
+    if checker_path:
+        command.append(f"--load-plugins={checker_path}")
+
+    return command
+
+
+def process_file(file_path):
```
Comment (ilaner) [inline demisto_sdk/commands/xsoar_linter/xsoar_linter.py:79]: types and docstrings

File: demisto_sdk/commands/xsoar_linter/xsoar_linter.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        errors += pattern.findall(log_str)
+    except subprocess.TimeoutExpired:
+        errors.append((f"Got a timeout while processing the following file: {str(file_path)}"))
+        return_code = 1
+    except Exception as e:
+        errors.append(f"Failed processing the following file: {str(file_path)}")
+        return_code = 1
+
+    errors_str = '\n'.join(errors) if errors else ''
+    return return_code, errors_str
```
Comment (ilaner) [inline demisto_sdk/commands/xsoar_linter/xsoar_linter.py:123]: retuned NamedTuple/dataclass

File: demisto_sdk/commands/xsoar_linter/xsoar_linter.py [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    command.append(str(file))
+
+    env.update(xsoar_linter_env)
+    env["PYTHONPATH"] = ":".join(str(path) for path in PYTHONPATH)
+    try:
+        process = subprocess.run(command, capture_output=True, env=env, timeout=60)
+        return_code = process.returncode
+        log_data = process.stdout
+        log_str = log_data.decode('utf-8')
+        pattern = re.compile(r'^/[^:\n]+:\d+:\d+: E\d+ .*$', re.MULTILINE)
```
Comment (ilaner) [inline demisto_sdk/commands/xsoar_linter/xsoar_linter.py:113]: make comment and use global var

**Review by ilaner — CHANGES_REQUESTED** [review]
Good job
See comments, and add test for the xsoar-linter command.

**Review by ilaner — APPROVED** [review]
Good job!

**Pull request merged**